app/components/GMap.jsx

```diff
@@ -1,81 +1,120 @@
 import React from 'react';
 import MapStyles from './MapStyles';
 import Script from 'react-load-script';
+import PropTypes from 'prop-types';
 
 export default class GMap extends React.Component {
   static get propTypes() {
     return {
-      center: React.PropTypes.objectOf(React.PropTypes.number).isRequired,
-      colors: React.PropTypes.objectOf(React.PropTypes.string),
-      message: React.PropTypes.string.isRequired,
-      findUserLocation: React.PropTypes.bool,
-      markerImage: React.PropTypes.string,
+      config: PropTypes.shape({
+        colors: PropTypes.objectOf(PropTypes.string),
+        icons: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+        initialCenter: PropTypes.objectOf(PropTypes.number),
+        initialZoom: PropTypes.number,
+        legend: PropTypes.bool,
+        markers: PropTypes.arrayOf(PropTypes.shape({
+          position: PropTypes.objectOf(PropTypes.number),
+          icon: PropTypes.string,
+          message: PropTypes.string
+        })),
+        snapToUserLocation: PropTypes.bool
+      })
     }
   }
 
   static get defaultProps() {
     return {
-        center: {
+      config: {
+        initialCenter: {
           lat: 29.975588,
-          lng: -90.102682 },
-        message: "A Message"
+          lng: -90.102682
+        },
+        initialZoom: 10
+      }
     }
   }
 
   constructor(props){
     super(props);
     this.state = {
-      zoom: 11,
-      infoWindowIsOpen: true
+      center: null
     };
   }
 
   loadMap() {
     if (this.state.scriptLoaded) {
-      // lets map autocenter on user's location (if the user enables it)
-      // which takes a while, so the map is created and goes to the initial center first
-      if (this.props.findUserLocation && navigator.geolocation) {
-        navigator.geolocation.getCurrentPosition( (position) => {
-          this.setState({
-            center: this.mapCenter(position.coords.latitude, position.coords.longitude)
-          });
-          this.moveMap("Got it!");
-        }, () => this.infoWindow.setContent("Couldn't find your location :("))
+      if (this.props.config && this.props.config.snapToUserLocation && navigator.geolocation) {
```

Comment:
nitpick, you are calling this.props.config a lot here, I would assign it to a `const` at the top of this method with `const {config} = this.props`